### DISCUSSION TITLE
tutorials/navigation/navigation_using_navigationservers

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#184] [User-contributed notes]
# tutorials/navigation/navigation_using_navigationservers

2D and 3D version of the NavigationServer are available as NavigationServer2D and NavigationServer3D respectively. Both 2D and 3D use the same NavigationServer with NavigationServer3D being the pri...

https://docs.godotengine.org/en/4.3/tutorials/navigation/navigation_using_navigationservers.html

**Comment by nezvers** [comment]
Need an example for creation of an obstacle on a NavigationServer.

**Reply by Calinou** [reply]
Please open an issue on the godot-docs repository for this: https://github.com/godotengine/godot-docs

**Comment by CodeSmith32** [comment]
RVO: "Reciprocal Velocity Object"

[Documentation of agent avoidance](https://docs.godotengine.org/en/stable/tutorials/navigation/navigation_using_navigationagents.html#navigationagent-avoidance)
([Older documentation mentioning "RVO"](https://docs.godotengine.org/en/4.0/tutorials/navigation/navigation_using_agent_avoidance.html#using-agent-avoidance))

**Comment by akaBaum** [comment]
The NavigationServer has a default map, which RID you can access with get_world_3d().navigation_map.
All Navigation nodes (NavigationAgent3D and NavigationRegion3D) use this NavigationMap if not changed.

So if you want to create your NavigationMesh from code, you have to add your NavigaitonRegion to the default NavigationMap, so that NaviagtionAgent3D nodes can find them. Or change the map the NaviagtionAgent3D is using.
